lib/chef/provider/package/chocolatey.rb

```diff
@@ -220,12 +220,18 @@ def cmd_args(include_source: true)
         # @return [Hash] name-to-version mapping of available packages
         def available_packages
           return @available_packages if @available_packages
+          choco_version = powershell_exec!("choco --version").result
 
           @available_packages = {}
           package_name_array.each do |pkg|
             available_versions =
               begin
-                cmd = [ "search", "-r", pkg ]
+                # Choco V2 uses 'Search' for remote repositories and 'List' for local packages
+                if choco_version.match?(/2/)
```

Comment:
would it make sense to check for version 1.x then use list else use search or possibly a begin, rescue to try each command. Just thinking about future compatibility with 3.x+.

Reply:
Agreed. 1.x is the exception at the moment

Reply:
```ruby
  def query_command
    return "list" if choco_version =~ /1\./
    "search"
  end
```